lib/Db/AssignedLabelsMapper.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * @copyright Copyright (c) 2020 Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @author Jakob Röhrl <jakob.roehrl@web.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+namespace OCA\Deck\Db;
+
+use OCP\AppFramework\Db\Entity;
+use OCP\IDBConnection;
+use OCP\IGroupManager;
+use OCP\IUserManager;
+
+class AssignedLabelsMapper extends DeckMapper  {
+	private $cardMapper;
+	private $userManager;
+	/**
+	 * @var IGroupManager
+	 */
+	private $groupManager;
+
+	public function __construct(IDBConnection $db, CardMapper $cardMapper, IUserManager $userManager, IGroupManager $groupManager) {
+		parent::__construct($db, 'deck_assigned_labels', AssignedLabels::class);
+		$this->cardMapper = $cardMapper;
+		$this->userManager = $userManager;
+		$this->groupManager = $groupManager;
+	}
+
+	/**
+	 *
+	 * @param $cardId
+	 * @return array|Entity
+	 */
+	public function find($cardId) {
+		$sql = 'SELECT * from `*PREFIX*deck_assigned_labels` where `card_id` = ?';
```

Comment:
Could be switched to QBMapper right away to get rid of the deprecated DeckMapper. See https://docs.nextcloud.com/server/18/developer_manual/app/storage/database.html?highlight=querybuilder#mappers for an example

Reply:
Feel free to do that in a follow up, but it would be good to not introduce new code that uses legacy functions right away :wink: